### ISSUE TITLE
Layout/ClosingParenthesisIndentation update leads to false indentation in keyword param edge case

### ISSUE THREAD

**Issue by kmoschcau** [rubocop-hq/rubocop#5889]
Hi, this might just be a small edge case and maybe not even desired layout according to the style guide, but if so it should be handled by another cop and not this one.

## Initial setup

```ruby
# Have a method definition with a keyword parameter, where you put the 
# default value on the next line, followed by another keyword and default
# value pair, or a double asterisk.
def foo(
  bar:
    'baz', **
)
end
```
As I said, this might not be a perfect layout in tune with the style guide, but right now there is nothing I found, that forbids this.

## Expected behavior

Rubocop should keep the closing parenthesis at exactly this indent and not complain.
That actually is the case, if you move the double asterisk or key-value-pair to a line on their own.

## Actual behavior

Rubocop wants to indent the closing parenthesis one further, on the same column as the "bar" symbol in the example.

## Steps to reproduce the problem

See the initial setup.

## RuboCop version

```
$ rubocop -V
0.56.0 (using Parser 2.5.1.0, running on ruby 2.3.5 x86_64-linux)
```

**Comment by antonzaytsev** [comment]
Can be fixed by https://github.com/rubocop-hq/rubocop/pull/6351
